File: src/main/java/me/centralhardware/znatoki/telegram/statistic/eav/PropertyDef.java
package me.centralhardware.znatoki.telegram.statistic.eav;

import com.fasterxml.jackson.databind.annotation.JsonDeserialize;
import me.centralhardware.znatoki.telegram.statistic.eav.jackson.TypeDeserializer;
import me.centralhardware.znatoki.telegram.statistic.eav.types.Type;

public record PropertyDef(
        @JsonDeserialize(using = TypeDeserializer.class)
        Type type,
        String name,
        String[] enumeration,
        Boolean isOptional,
        Boolean isUnique
){}

File: src/main/kotlin/me/centralhardware/znatoki/telegram/statistic/eav/PropertyDef.kt
package me.centralhardware.znatoki.telegram.statistic.eav

import kotlinx.serialization.Serializable
import me.centralhardware.znatoki.telegram.statistic.eav.json.TypeAdapter
import me.centralhardware.znatoki.telegram.statistic.eav.types.Type

@Serializable
data class PropertyDef(
    @Serializable(with = TypeAdapter::class)
    val type: Type,
    val name: String,
    val enumeration: List<String> = listOf(),
    val isOptional: Boolean,
    val isUnique: Boolean = false
)